notebooks/concepts/Specviz from splot concept notebook.ipynb

```diff
@@ -0,0 +1,3636 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This notebook's goal is to present a concept notebook of specviz, using workflows from the [IRAF splot tool](https://iraf.net/irafhelp.php?val=splot&help=Help+Page).  The feature/workflow list for splot came from a [trello board](https://trello.com/b/glMEUlE3/iraf-splot-feature-list), but the features are copied here to frame this notebook."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This notebook is organized around broad themes of planned `specviz` features, and how they address specific SPLOT functionality.  All SPLOT functionality is listed here, although at the end several features are listed as \"not to be implemented\".\n",
+    "\n",
+    "Also note that mostt feattures in this notebook are divided into two sections: \n",
+    "1. A specification of how this can/should be accomplished \"in the notebook\".  It is important to recognize that \"in the notebook\" assumes there is an active specviz session somewhere in the notebook that is used as the \"hub\" for operations, even when the operations are all Python functions that do not directly require interactivity. \n",
+    "2. The desktop interface - this is more focused on \"quick look\" type behavior."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Loading data/creating specviz:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Spectra to be loaded in specviz should be compatible with `Spectrum1D`, which should be viewed as the prime \"input\" to `specviz`.  \n",
+    "\n",
+    "1.  This means that in the notebook, `Spectrum1D` loaders can be used directly:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import specutils\n",
+    "\n",
+    "spec_url = 'https://dr14.sdss.org/optical/spectrum/view/data/format=fits/spec=lite?plateid=1323&mjd=52797&fiberid=12'\n",
+    "spec = specutils.Spectrum1D.read(spec_url)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "    This spectrum should then be natively read by the specviz helper object. Note that this should automatically select the `.flux` part of the spectrum in the viewer, *not* require the user to do something to load that spectrum."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import jdaviz\n",
+    "\n",
+    "specviz = jdaviz.Specviz()\n",
```

Comment:
This is alluding to a proposed application-specific functionality where instances of specviz, mosviz, and cubeviz will use wrapper classes that take in the default application instance:

```python
class SpecVizApp:
    def __init__(self):
        self._app = Application(configuration='specviz')
    
    def some_specviz_specific_method(self):
        ...
```

You can see this in #149.

Reply:
There are myriad benefits to using a helper class instead of subclassing: we want composition over inheritance because the state of the generic application should not be tied to the specific instance a user is using. We only want the helper classes as a means for providing specific functionality, but not such that the helper classes maintain their own state of the application object as a whole. This allows for transitioning between application instances simply by wrapping the generic application object in a new helper class and (and possibly reseting the configuration).

This should not really impact the MAST side, as the added benefit of a helper class is mainly for notebook access (e.g. custom methods that do specviz-/cubeviz-/mosviz-specific data translations like defaultly retrieving `Spectrum1D` objects when calling `get_data_from_viewer`, or in cases where we can guarantee that there's a `spectrum-viewer` viewer for instance).

Please see #104 for a more thorough break down of the reasoning.

Reply:
Thanks for the clarification on the helper class.  That makes sense.  

It will impact MAST but I can't yet say how much.  Currently we load a default Application and then dynamically reconfigure and customize the application based on the input file.  See #158,  I can also provide our notebook use case.  If MAST wants to take advantage of the tailored helper class functionality (and we do) they might limit our ability to do that.  We will need to implement more complicated configuration logic.  Now that #149 is merged, I will take a deeper look at it, and see what happens when I try to integrate with what we need.

Reply:
This is the "helper" class discussed in #104 (particularly https://github.com/spacetelescope/jdaviz/issues/104#issuecomment-624740003).  I'll add a point specifically pointing that out, though!